openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -155,7 +155,7 @@ def run(self, problem):
             self.sparsity[name] = self.indep_list
 
         # Calculate and save gradient for any linear constraints.
-        lcons = self.get_constraints(lintype='linear').values()
+        lcons = self.get_constraints(lintype='linear').keys()
```

Comment:
I think keys() returns an iterator in py3.  Not sure that len() of an iterator will work...